actionmailer/lib/action_mailer/base.rb

```diff
@@ -944,6 +831,115 @@ def self.supports_path?
       false
     end
 
+  private
+
+    def default_values
+      @_default_values ||= self.class.default.map do |key, value|
```

Comment:
This don't need memoization and also it is harder to understand now. I'd just move it closed to where it is being used with the previous implementation.